app/src/androidTest/java/com/lightricks/feedexercise/data/FeedRepositoryTest.kt

```diff
@@ -1,13 +1,111 @@
 package com.lightricks.feedexercise.data
 
+import android.content.Context
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.Observer
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.database.FeedItemDao
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.TemplatesMetadata
+import com.lightricks.feedexercise.ui.feed.FeedViewModel
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import io.reactivex.Single
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
 import org.junit.runner.RunWith
-import java.util.concurrent.CountDownLatch
-import java.util.concurrent.TimeUnit
+import java.io.IOException
 
 @RunWith(AndroidJUnit4::class)
 class FeedRepositoryTest {
    //todo: add the tests here
+
+    private lateinit var userDao: FeedItemDao
+    private lateinit var db: FeedDatabase
+    private lateinit var appContext: Context
+    private lateinit var jsonContent: String
+    private lateinit var serviceResponse: TemplatesMetadata
+
+    @Before
+    fun createDb() {
+        appContext = ApplicationProvider.getApplicationContext<Context>()
+        db = Room.inMemoryDatabaseBuilder(
+            appContext, FeedDatabase::class.java).build()
+        userDao = db.feedItemDao()
+    }
+
+    @Before
+    fun getJsonStr(){
+        jsonContent = appContext.assets
+            .open("get_feed_response.json")
+            .bufferedReader()
+            .use{ it.readText() }
+        val moshi = Moshi.Builder()
+            .add(KotlinJsonAdapterFactory())
+            .build()
+        val adapter = moshi.adapter<TemplatesMetadata>(TemplatesMetadata::class.java)
+        serviceResponse = adapter.fromJson(jsonContent)!!
+    }
+
+    @After
+    @Throws(IOException::class)
+    fun closeDb() {
+        db.close()
+    }
+
+    @get:Rule
+    var instantExecutorRule = InstantTaskExecutorRule()
+
+    @Test
+    fun combinedTest() {
```

Comment:
Can you split this into 3 separate tests? 
For one thing, test names tell us the story (the expected behavior). At this point I don't know what this test is supposed to check.
Also, fixing short tests is usually simpler.